templates/archives/stories/debugging.jinja2

```diff
@@ -1,38 +1,60 @@
-    was not necessarily the bottleneck here.  <br />
+    Given the emphasis now placed on the speed of computers and the incredible gains of <a href="https://en.wikipedia.org/wiki/Moore%27s_law">
+    Moore’s Law</a>, one might presume that the primary issue with the computers of the 1950’s was that they were
+    slow.  Computers were already much faster and more reliable than any human being could hope to be.  Lack of
+    processing power was not necessarily the bottleneck here.  <br />
+    <figure class="figure float-right" style="height:25%;width:25%">
+        <img src="{{ static('img/stories/debugging.png') }}"
+            class="figure-img img-fluid rounded"
+            alt="Part of source for the Fortran II compiler for IBM 704." />
+        <figcaption class="figure-caption">Part of source for the Fortran II compiler.</figcaption>
+    </figure>
     <br />
     One major problem that Morse identified was the input and output.  Computers back then didn’t have a screen and a
     keyboard.  Programs had to be converted to punch card by human, which is bound to introduce errors.  And
-    according to Morse, it could often take much more time for the computer to punch out the answer than it takes to
-    run the program!  <br />
+    if the output required more than printing a few numbers, the results would have to be printed to magnetic tape
+    (which blocked the computer), then converted to punched paper tape by a slow mechanical device, and then
+    converted into human-readable format by an automatic typewriter.  Or, one could display the results on a
+    cathode-ray tube, take photographs of each screen, and then manually transcribe the photographs by hand to a more
+    legible form.<br />
     <br />
-    This latter issue was especially problematic for debugging.  In modern times, debugging is simplified by printing
-    out values of variables or use of more advanced tools like gdb that allow one to view the entire state of the
-    program as it runs.  Not so in the 1950’s.  The only way to figure out what was happening inside a defective
-    program was to halt the program prematurely and slowly wait for the computer to print out all the values, and
-    hope that one of them will point to the origin of the bug.  Because the output was so slow, Philip Morse
-    suggested that one try to write one’s entire program in one go while minimizing the number of output values.
-    Modern coding practice suggests that this is a recipe for disaster, but what else can you do?  Everyone had to
-    share the computer, meaning you couldn’t hog the machine for several hours while you wrote the program chunk by
-    chunk.  <br />
+    This input/output issue was especially problematic for debugging.  In modern times, one of the first things the
+    beginner CS student learns is to inject <code>print</code> statements all throughout their code to figure
```

Comment:
I would write out "computer science" instead of the "CS" abbreviation to make sure it's more accessible to all readers.